lib/redux/actions/async.js

```diff
@@ -428,19 +431,31 @@ export function doVersionCheck() {
 
 export function setTargetTimezone(userId, timezoneName) {
   return (dispatch, getState) => {
+    const { allUsers } = getState();
     const { api } = services;
     dispatch(syncActions.updateProfileRequest());
-    api.user.updateProfile(userId, {
+    let updates = {
       patient: {
         targetTimezone: timezoneName
       }
-    }, (err, profile) => {
+    };
+    api.user.updateProfile(userId, updates, (err, profile) => {
       // suppress unauthorized error until custodial vs normal permissions are ironed out
       // TODO: remove conditional when perms are finalized or accounts are converted
-      if(err && _.get(err,'status') !== 401){
-        dispatch(syncActions.updateProfileFailure(err));
+      if (err){
+        if (_.get(err,'status') !== 401) {
+          dispatch(syncActions.updateProfileFailure(err));
+        } else {
+          let profile = allUsers[userId];
+          let patient = _.get(profile, 'patient', {});
+          let patient_updates = _.get(updates, 'patient', {});
+          let new_patient = _.assign(profile.patient, patient_updates);
```

Comment:
I don't believe you're using this `new_patient` either. This whole sequence is really hard to understand. Can you pare it down?